landoapi/hgexports.py

```diff
@@ -605,3 +616,144 @@ def run_diff_checks(self) -> list[str]:
                 issues.append(issue)
 
         return issues
+
+
+@dataclass
+class PushCheck:
+    """Provides an interface to implement push-wide checks."""
```

Comment:
We do already have a concept of "push-to-Lando", which will expand as Git support is added since try/automation/sheriffs will all need to submit sets of patches in one go, which write to the pushlog as a single entry. Most of our users are familiar with the idea of a push being a set of commits/changesets that are sent to a repo as a single unit, so I fear introducing a separate internal naming for this concept might be a bit confusing. If we do want to avoid using `Push...`, I would vote for `PatchCollectionCheck`, since we already have a `StackAssessor` which does something different from this API.

Reply:
I am OK with `PatchCollectionCheck`, makes sense to avoid `StackCheck` since it is too similar to `StackAssessor`. I guess we would also rename `PushAssessor` to `PatchCollectionAssessor`.